mitiq/executor/executor.py

```diff
@@ -150,16 +150,7 @@ def evaluate(
                 for circuit_with_measurements in observable.measure_in(circuit)
             ]
             result_step = observable.ngroups
-        elif (
-            observable is not None
-            and self._executor_return_type not in MeasurementResultLike
-            and self._executor_return_type not in DensityMatrixLike
-        ):
-            raise ValueError(
-                """Executor and observable are not compatible. Executors
-                returning expectation values as float must be used with
-                observable=None"""
-            )
```

Comment:
Could you give more details why you had to remove this check?
What is the part of the code that is incompatible with this?

Reply:
> This looks promising! Do you think we should keep both the original PEC-based learning method and this new unmitigated learning method? Or are we eventually going to choose one of the two?

I think we should choose one or the other- testing and maintaining both is probably more overhead than it's worth. I implemented this "unmitigated" version because I thought the original, mitigated loss function had reached somewhat of a dead end. In working out the unmitigated version I found an approach that also worked for the mitigated loss function. We can keep this open as a backup option until we merge #1358 .

Reply:
> Could you give more details why you had to remove this check? What is the part of the code that is incompatible with this?

If we don't test on Qiskit, removing this check isn't needed as far this PR is concerned, but I think it might be a problem in general. When using `qiskit_utils.execute`, `execute_with_noise`, etc, `executor.evaluate` will raise an error that the executor and observable are not compatible. Shouldn't the compatibility check be done when applying error mitigation, e.g. in `execute_with_pec`? Btw, I was the one who added the compatibility check to `evaluate` in #1198.

Reply:
Looking better your qiskit test, the fix is probably to set `observable=None` in `learn_noise_parameters_from_unmitigated_data`. Indeed, after you fix the observable in the executor, it becomes and observable-less one (i.e. it returns a float) and so it must be used with `observable=None` in all Mitiq techniques.